tests/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml

```diff
@@ -881,6 +882,51 @@
 
 
   # ============================================================
+
+  - name: Delete NAT gateway
+    ec2_vpc_nat_gateway:
+      nat_gateway_id: '{{ nat_gateway_id }}'
+      state: absent
+      wait: yes
+    register: delete_nat_gateway
+
+  # ============================================================
+
+  - name: Create new NAT gateway with connectivity_type = private - CHECK_MODE
+    ec2_vpc_nat_gateway:
+      subnet_id: '{{ subnet_id }}'
+      connectivity_type: 'private'
+      wait: yes
+    register: create_ngw
+    check_mode: yes
+
+  - name: Assert creation happened (expected changed=true) - CHECK_MODE
+    assert:
+      that:
+      - create_ngw.changed
```

Comment:
overall LGTM! nit. for `check_mode` assertions, I like to add assertion for resource_actions to make sure it did not really create the resource.
so something like, but up to you.
`- '"ec2:CreateNatGateway" not in create_ngw.resource_actions'`